### ISSUE TITLE
New actions to better support integration with digital signature solution

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#2163]
This calls for adding new actions and maybe constructs to the language we're using to describe processes, and this to make it easier for Orbeon Forms to talk to a "signature server" and an "signature applet". The sequence of actions will look as follows:

![screenshot 2015-04-03 at 11 14 06 pm](https://cloud.githubusercontent.com/assets/106961/6991674/30acacae-da57-11e4-9a4c-d766df33ac37.png)

1. The end user clicks on the sign button. This is a standard Form Runner button, setup to run a process.
2. The process starts by generating the PDF for the form (new action to trigger the existing PDF generation capability), and send it to the signature server (new action).
3. The signature server stores the PDF and associates it with an id (`sig_id`), which it returns.
4. The process continues by extracting the id from the response (new action or construct), and calls JavaScript code passing that id (new action).
5. The JavaScript runs an applet, which, with the `sig_id` can talk to the signature server, get the PDF, show it, and upon the user signing the form, send the digital signature to the signature server. That digital signature could also be stored in a form field, if necessary.

[+1 from customer](https://basecamp.com/1721271/projects/2291082/messages/23480960)

[1]: https://github.com/orbeon/orbeon-forms/wiki/Form-Runner-~-Buttons-and-Processes

**Comment by ebruchez** [comment]
In 2015 you would think that an applet would no longer be needed and that the signature could be done in JavaScript.

**Comment by avernet** [comment]
@ebruchez Yes, it is possible that some of those products moved from an applet to an extension  as a way to communicate with code installed on the machine (which in turn talks to a card reader). E.g. as WebEx did. But this, from Orbeon Forms' perspective, shouldn't make a difference as in both cases it comes down to calling JavaScript.

**Comment by ebruchez** [comment]
+1 from evaluator to have the ability to digitally sign the data and PDF when saving the data, to have the assurance that the data is not modified at a later time.
